specification/fabric/Microsoft.Fabric.Management/models.tsp

```diff
@@ -0,0 +1,176 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@azure-tools/typespec-azure-core";
+
+using TypeSpec.Http;
+using Azure.Core;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Foundations;
+using OpenAPI;
+
+namespace Microsoft.Fabric;
+
+interface Operations extends Azure.ResourceManager.Operations {}
+
+@doc("The Microsoft Fabric capacity properties.")
+model FabricCapacityProperties {
+  @doc("The current deployment state of Microsoft Fabric resource. The provisioningState is to indicate states for resource provisioning.")
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+
+  @doc("The current state of Microsoft Fabric resource. The state is to indicate more states outside of resource provisioning.")
+  @visibility("read")
+  state: State;
+
+  @doc("The capacity administration")
+  administration: CapacityAdministration;
+}
+
+@doc("Represents the SKU name and Azure pricing tier for Microsoft Fabric capacity resource.")
+model RpSku {
+  @doc("The name of the SKU level.")
+  name: string;
+
+  @doc("The name of the Azure pricing tier to which the SKU applies.")
+  tier: RpSkuTier;
+}
+
+@doc("The name of the Azure pricing tier to which the SKU applies.")
+union RpSkuTier {
+  string,
+
+  @doc("Fabric tier")
+  Fabric: "Fabric",
+}
+
+@doc("The administration properties of the Fabric capacity resource")
+model CapacityAdministration {
+  @doc("An array of administrator user identities.")
+  members: Array<string>;
+}
+
+@doc("The resource provisioning state.")
+union ProvisioningState {
+  ResourceProvisioningState,
+
+  /** Resource is deleting */
+  Deleting: "Deleting",
+
+  /** Resource is provisioning */
+  Provisioning: "Provisioning",
+
+  /** Resource is updating */
+  Updating: "Updating",
+
+  string,
+}
+
+/** The resource provisioning state. */
+union State {
+  ResourceProvisioningState,
```

Comment:
The State set includes Succeeded, Failed, Canceled, Provisioning, Updating, and Deleting, which are all duplicated from ProvisioningState. This makes a pretty confusing state model. The duplicated states make many combinations whose meaning is unclear.

Can you pare down the number of states?

I would expect the set of state values allowed by each of these two state variables to be disjoint. IOW, the "provisioningState" variable changes value only while the resource is being created or updated (Creating/Updating => Succeeded/Failed/Canceled) and the "state" variable changes value only after the resource begins operation (Inactive/Unavailable (during creation) => Active, followed by operational states (e.g. Suspending => Suspended or Pausing => Paused, etc.).

Reply:
@xinzhizou kindly request your input here

Reply:
Sorry, I wanted to get clarification on the state names. Updating makes sense to me as a transition state for PATCH and PUT of an existing resource. Deleting makes sense as a transition state for DELETE of an existing resource. That means that Provisioning would only happen on PUT of a newly created resource, which is why I think Creating makes is a better state name. Thoughts?

Reply:
**ProvisioningState** is used for async operation contract documented here: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md

Our RP implementation supports 5 values for it: "Deleting", "Succeeded", "Failed", "Provisioning", "Updating"
For code pls see https://powerbi.visualstudio.com/MWC/_git/aspaas?path=/Apps/FrontendService/ASPaaS.FrontEnd.Service/Global/FabricRpProvisioningStates.cs&version=GBmaster

**State** is used to indicate the exact state of our RP resource capacities, so it has more specific values such as "Pausing", "Resuming". When capacity is "Pausing", the corresponding provisioningState for async operation is "Updating". For all capacity states, pls see https://powerbi.visualstudio.com/MWC/_git/aspaas?path=/Apps/FrontendService/ASPaaS.FrontEnd.Service/Global/FabricRpResourceStates.cs&version=GBmaster

@badeamarjieh in my original PR, I used "ResourceStateKV" as variable name to make it clear that it's for RP resource and different from the other provisioningState. Can you please double check my PR and update?

Reply:
Yes, this makes sense to me (although I don't have access to those links). I don't think the ResourceProvisioningState states should be replicated in the State states.